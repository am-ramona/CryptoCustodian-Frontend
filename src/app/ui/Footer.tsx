import React from 'react'
import type { NextPage } from 'next'

const Footer: NextPage = () => {
	return (
		<footer className="grid h-20 text-center">
			<p className="place-self-center text-[15px] font-medium pt-3">&copy; 2024 Crypto Custodian</p>
	  	</footer>
	);
}

export default Footer