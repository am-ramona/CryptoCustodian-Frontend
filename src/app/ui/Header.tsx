import React from  'react'
import type { NextPage } from 'next'
import Logo from './Logo'

const Header: NextPage = () => {
  return (
    <header className='w-full h-20'>
      <Logo />
    </header>
  );
};

export default Header