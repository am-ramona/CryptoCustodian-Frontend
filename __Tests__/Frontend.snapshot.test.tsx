import React from 'react';
import { create } from 'react-test-renderer';
import Header from '../src/app/ui/Header';
import Footer from '../src/app/ui/Footer';
import Navigation from '../src/app/ui/Nav';

describe('Header Component Snapshot', () => {
  test('matches the snapshot', () => {
    const tree = create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Footer Component', () => {
    test('matches the snapshot', () => {
      const tree = create(<Footer />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Navigation Snapshot', () => {
    test('matches the snapshot', () => {
      const tree = create(<Navigation show={true} setter={() => {}} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  
  
  
