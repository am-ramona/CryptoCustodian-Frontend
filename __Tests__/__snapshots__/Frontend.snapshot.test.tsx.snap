// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches the snapshot 1`] = `
<footer
  className="grid h-20 text-center"
>
  <p
    className="place-self-center text-[15px] font-medium pt-3"
  >
    Â© 2024 Crypto Custodian
  </p>
</footer>
`;

exports[`Header Component Snapshot should match the snapshot 1`] = `
<header
  className="w-full h-20"
>
  <a
    className="undefined text-#232323"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    <h1
      className="font-mono font-normal text-lg p-6 cursor-pointer"
    >
      Crypto Custodian
    </h1>
  </a>
</header>
`;

exports[`Navigation Snapshot matches the snapshot 1`] = `
[
  <div
    className="w-fit h-full transition-[margin-left] ease-in-out duration-500 fixed md:static top-[80px] bottom-0 left-0 z-40 ml-0 h-full bg-offWhite text-blackLight/75"
  >
    <div
      className="flex flex-col"
    >
      <a
        className="flex gap-2 [&>*]:my-auto text-md pl-6 py-3 border-b-[1px] border-b-white/10 text-inherit/75 hover:text-inherit"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="text-xl flex [&>*]:mx-auto text-green"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable="false"
            height={16}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 17v8H6V17H16m0-2H6a2 2 0 00-2 2v8a2 2 0 002 2H16a2 2 0 002-2V17a2 2 0 00-2-2zM27 6v5H17V6H27m0-2H17a2 2 0 00-2 2v5a2 2 0 002 2H27a2 2 0 002-2V6a2 2 0 00-2-2zM27 17v5H22V17h5m0-2H22a2 2 0 00-2 2v5a2 2 0 002 2h5a2 2 0 002-2V17a2 2 0 00-2-2zM11 6v5H6V6h5m0-2H6A2 2 0 004 6v5a2 2 0 002 2h5a2 2 0 002-2V6a2 2 0 00-2-2z"
            />
          </svg>
        </div>
        <div
          className="font-mono font-normal"
        >
          Dashboard
        </div>
      </a>
      <a
        className="flex gap-2 [&>*]:my-auto text-md pl-6 py-3 border-b-[1px] border-b-white/10 text-inherit/75 hover:text-inherit"
        href="/assetOverview"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="text-xl flex [&>*]:mx-auto text-green"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable="false"
            height={16}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 21H26V26H24zM20 16H22V26H20zM11 26a5.0059 5.0059 0 01-5-5H8a3 3 0 103-3V16a5 5 0 010 10z"
            />
            <path
              d="M28,2H4A2.002,2.002,0,0,0,2,4V28a2.0023,2.0023,0,0,0,2,2H28a2.0027,2.0027,0,0,0,2-2V4A2.0023,2.0023,0,0,0,28,2Zm0,9H14V4H28ZM12,4v7H4V4ZM4,28V13H28.0007l.0013,15Z"
            />
          </svg>
        </div>
        <div
          className="font-mono font-normal"
        >
          Assets Overview
        </div>
      </a>
      <a
        className="flex gap-2 [&>*]:my-auto text-md pl-6 py-3 border-b-[1px] border-b-white/10 text-inherit/75 hover:text-inherit"
        href="/positionsAndHoldings"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="text-xl flex [&>*]:mx-auto text-green"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable="false"
            height={16}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 24H28V26H4zM26 18H6V14H26v4m2 0V14a2 2 0 00-2-2H6a2 2 0 00-2 2v4a2 2 0 002 2H26a2 2 0 002-2zM4 6H28V8H4z"
            />
          </svg>
        </div>
        <div
          className="font-mono font-normal"
        >
          Positions and Holdings
        </div>
      </a>
      <a
        className="flex gap-2 [&>*]:my-auto text-md pl-6 py-3 border-b-[1px] border-b-white/10 text-inherit/75 hover:text-inherit"
        href="/aprAndPerformanceMetrics"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="text-xl flex [&>*]:mx-auto text-green"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable="false"
            height={16}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M26,22a3.9553,3.9553,0,0,0-2.0193.5667L20.167,18.7529a4.9654,4.9654,0,0,0,0-5.5058L22.714,10.7A3.0284,3.0284,0,1,0,21.3,9.286l-2.547,2.547a4.9654,4.9654,0,0,0-5.5058,0L9.4332,8.0191A3.9553,3.9553,0,0,0,10,6a4,4,0,1,0-4,4,3.9553,3.9553,0,0,0,2.0191-.5668l3.8139,3.8139a4.9654,4.9654,0,0,0,0,5.5058L8.0192,22.5668A3.9556,3.9556,0,0,0,6,22a4,4,0,1,0,4,4,3.9553,3.9553,0,0,0-.5667-2.0192l3.8138-3.8138A4.9686,4.9686,0,0,0,15,20.8989v3.2848a3,3,0,1,0,2,0V20.8989a4.9686,4.9686,0,0,0,1.7529-.7319l3.8138,3.8137A3.9553,3.9553,0,0,0,22,26a4,4,0,1,0,4-4ZM16,13a3,3,0,1,1-3,3A3.0033,3.0033,0,0,1,16,13ZM4,6A2,2,0,1,1,6,8,2.002,2.002,0,0,1,4,6ZM6,28a2,2,0,1,1,2-2A2.0023,2.0023,0,0,1,6,28Zm20,0a2,2,0,1,1,2-2A2.0027,2.0027,0,0,1,26,28Z"
            />
          </svg>
        </div>
        <div
          className="font-mono font-normal"
        >
          APR and Performance Metrics
        </div>
      </a>
      <a
        className="flex gap-2 [&>*]:my-auto text-md pl-6 py-3 border-b-[1px] border-b-white/10 text-inherit/75 hover:text-inherit"
        href="/clientManagement"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="text-xl flex [&>*]:mx-auto text-green"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable="false"
            height={16}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M26 30h-2v-5c-.0033-2.7601-2.2399-4.9967-5-5h-6c-2.7601.0033-4.9967 2.2399-5 5v5h-2v-5c.0045-3.8641 3.1359-6.9955 7-7h6c3.8641.0045 6.9955 3.1359 7 7v5zM22 6v4c0 1.1025-.8972 2-2 2h-1c-.5522 0-1 .4478-1 1s.4478 1 1 1h1c2.2056 0 4-1.7944 4-4v-4h-2zM16 16c-3.8599 0-7-3.1401-7-7S12.1401 2 16 2c1.9885 0 3.8901.8503 5.2173 2.3329l-1.4902 1.334c-.9482-1.0593-2.3066-1.6669-3.7271-1.6669-2.7571 0-5 2.243-5 5s2.2429 5 5 5v2z"
            />
          </svg>
        </div>
        <div
          className="font-mono font-normal"
        >
          Client Management
        </div>
      </a>
      <a
        className="flex gap-2 [&>*]:my-auto text-md pl-6 py-3 border-b-[1px] border-b-white/10 text-inherit/75 hover:text-inherit"
        href="/transactionAndCompliance"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="text-xl flex [&>*]:mx-auto text-green"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable="false"
            height={16}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="19"
              cy="27"
              r="1"
            />
            <path
              d="M29 31h-13c-1.1028 0-2-.8972-2-2v-4c0-1.1028.8972-2 2-2h13c1.1028 0 2 .8972 2 2v4c0 1.1028-.8972 2-2 2zm-13-6v4h13v-4h-13zM29 21h-13c-1.1028 0-2-.8972-2-2v-4c0-1.1028.8972-2 2-2h13c1.1028 0 2 .8972 2 2v4c0 1.1028-.8972 2-2 2zm-13-6v4h13v-4h-13z"
            />
            <path
              d="m12,27.2998c-4.6531-1.6523-8-6.0869-8-11.2998,0-6.6167,5.3833-12,12-12,4.8311,0,8.9937,2.876,10.8945,7h2.1665c-2.0188-5.2544-7.1045-9-13.061-9-7.7197,0-14,6.2803-14,14,0,6.3291,4.2246,11.6846,10,13.4102v-2.1104Z"
            />
          </svg>
        </div>
        <div
          className="font-mono font-normal"
        >
          Transaction and Compliance
        </div>
      </a>
    </div>
  </div>,
  <div
    className="flex md:hidden fixed top-0 right-0 bottom-0 left-0 bg-blackLight/50 z-30"
    onClick={[Function]}
  />,
]
`;
